File: Java-Core/src/main/java/prompto/value/Image.java
package prompto.value;

import java.io.IOException;

import com.fasterxml.jackson.core.JsonGenerator;

import prompto.error.PromptoError;
import prompto.error.ReadWriteError;
import prompto.runtime.Context;
import prompto.type.ImageType;

public class Image extends Binary {

	public Image() {
		super(ImageType.instance());
	}
	
	public Image(String mimeType, byte[] dataBytes) {
		super(ImageType.instance(), mimeType, dataBytes);
	}

	public static Image fromResource(String path) throws IOException {
		Image image = new Image();
		Binary.fromResource(image, path);
		return image;
	}

	@Override
	public void toJson(Context context, JsonGenerator generator) throws PromptoError {
		try {
			generator.writeString("empty.jpg");
		} catch(IOException e) {
			throw new ReadWriteError(e.getMessage());
		}
	}
}

File: Core/src/main/resources/prompto/intrinsic/Image.js
function Image() {
	this.mimeType = null;
	this.url = null;
	this.file = null;
	return this;
}

Image.fromJSON = function(value) {
	var image = new Image();
	image.mimeType = value.mimeType;
	image.url = value.url;
	return image;
}

Image.fromFile = function(file) {
	var image = new Image();
	image.mimeType = file.type;
	image.file = file;
	return image;
}